### ISSUE TITLE
[E] [TRT] CUDA initialization failure with error 35.

### ISSUE THREAD

**Issue by shubhamgajbhiye1994** [wang-xinyu/tensorrtx#685]
Please check your CUDA installation:  http://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html

I am tried ultralytics/yolov5:v5.0   , nvcr.io/nvidia/pytorch:latest , ultralytics/yolov5:latest to generate engine file for yolov5 small model but I am getting cuda initialization error.
can you please guide what cuda version is need and ist supporting lib like tensorrt and other stuff to generate tensort file.

yolov5:v5.0 has cuda 11.0 (preinstalled)
yolov5:latest has cuda 11.3 (preinstalled)
nvcr.io/nvidia/pytorch:latest:11.3 (preinstalled)

Please , if you have link to old docker images please share with me

**Comment by wang-xinyu** [comment]
Firstly, you need install only one CUDA. And then install pytorch and tensorrt which depend on that CUDA version.

**Comment by shubhamgajbhiye1994** [comment]
Yeah! but the thing is that docker container is already having cuda installed and repespective lib installed but it still giving error.
can be please give me more which cuda version to be used

**Comment by shubhamgajbhiye1994** [comment]
@wang-xinyu  - Do we need to use gpu to convert  yolov5s.pts to trt file ?

**Comment by wang-xinyu** [comment]
wts to trt, of course need GPU, trt can only run on GPU.

**Comment by AhmedHisham1** [comment]
Did you solve this? I am facing the same issue. 
However, the `[E] [TRT] CUDA initialization failure with error 35` error occurs only when I am running the `./yolov5 -s ./yolov5.wts ./yolov5.engine s` command within the Dockerfile, but when I run this command inside the image I am using, the command executes properly and the engine file is generated correctly.

**Comment by shubhamgajbhiye1994** [comment]
@AhmedHisham1  have you resolved the same issue which I am facing
, I am unable to resolved it

**Comment by AhmedHisham1** [comment]
Yes. You just need to specify nvidia as the default runtime for all docker images. Because the intermediate layers/images that are being created for each run command will not be able to use nvidia by default.
So inside `/etc/docker/daemon.json` add this line `"default-runtime": "nvidia"`.

**Comment by shubhamgajbhiye1994** [comment]
@AhmedHisham1  can you please share your docker file code

**Comment by herbat** [comment]
Having the same issue with almost the exact same setup (except I'm using a slightly different docker base from NVIDIA, https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tensorrt). Have you checked the CUDA version installed in your base system? If so, what versions do `nvidia-smi` and `nvcc -V`, and `cat /etc/cuda/version.txt` return?

**Comment by dragen1860** [comment]
have the same error 35. any update? help.
